src/components/Pages/SetupPage/SetupPage.tsx

```diff
@@ -395,6 +398,24 @@ export const SetupPage = () => {
     })()
   }, [redirect, navigate, account, identity, isConnecting, initializedFlags, flags])
 
+  useEffect(() => {
+    const url = config.get('REFERRAL_SERVER_URL')
+    const referrer = urlSearchParams.get('referrer')
+
+    if (url && identity && referrer && EthAddress.validate(referrer)) {
+      fetch(`${url}/referral-progress`, {
+        method: 'POST',
+        headers: {
+          contentType: 'application/json'
+        },
+        body: JSON.stringify({
+          referrer
+        }),
+        identity
+      })
+    }
```

Comment:
If I'm not wrong, un-handled catches will show up in Sentry. As a user could get more than once into this site, would you mind catching and logging any potential errors?